src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -325,12 +334,22 @@ async def setup_pytest_for_target(
     cache_scope = (
         ProcessCacheScope.PER_SESSION if test_subsystem.force else ProcessCacheScope.SUCCESSFUL
     )
+
+    xdist_concurrency: int = 0
+    if pytest.xdist_enabled and not request.is_debug:
+        concurrency = request.field_set.xdist_concurrency.value
+        if concurrency is None:
+            contents = await Get(DigestContents, Digest, field_set_source_files.snapshot.digest)
+            concurrency = _count_pytest_tests(contents)
+        xdist_concurrency = concurrency
+
     process = await Get(
         Process,
         VenvPexProcess(
             pytest_runner_pex,
             argv=(
                 *(("-c", pytest.config) if pytest.config else ()),
+                *("-n", "{pants_concurrency}" if xdist_concurrency else "0"),
```

Comment:
Nit, maybe this so we don't add unnecessary args?

```suggestion
                *(("-n", "{pants_concurrency}") if xdist_concurrency else ()),
```

Reply:
I _think_ this might actually be necessary to explicitly disable xdist even though it's in the venv - will double-check

Reply:
I was wrong 😄 tests pass with this, will include it in the push with other changes